### DISCUSSION TITLE
Is it possible to open a qutebrowser session listening for debugging on some port?

### DISCUSSION THREAD

**Discussion by Aster89** [qutebrowser/qutebrowser#7152] [Q&A] [answered]
I can launch Chromium like this
```
chromium --remote-debugging-port=9229
```
and then a debugger can be attached to that port. Can such a thing be done with qutebrowser?

**Comment by The-Compiler** [comment, marked as answer]
That flag seems to work when passed via `--qt-flag remote-debugging-port=9229` - or at least I see the "DevTools listening on ws://127.0.0.1:9229/devtools/browser/..." message.

**Reply by Aster89** [reply]
I can confirm that if I open qutebrowser _only like that_, then I am indeed able to attach a debugger to it.

However, if I have an already open qutebrowser session open, and then I run that command, I see the message _Opening in existing instance_, and a new window opens up; however, I'm not able to attach the debugger to it.

Not sure if it can be relevant, but I get an error message like the following when trying to attach the debugger:
 > Cannot connect to runtime process, timeout after 10000 ms - (reason: Cannot connect to the target: connect ECONNREFUSED 127.0.0.1:9229).

---

This means that whenever I want to debug, I need to close the current qutebrowser session (which I haven't opened with the command you suggested), and open a new one from terminal with the command you suggested.

**Reply by emanuele6** [reply]
> This means that whenever I want to debug, I need to close the current qutebrowser session (which I haven't opened with the command you suggested), and open a new one from terminal with the command you suggested.

You can start qutebrowser with `-T` to not use the main qutebrowser instance: `qutebrowser -T --qt-flag remote-debugging-port=9229`.
